File: 55. Jump Game/55. Jump Game.ts
function canJump(nums: number[]): boolean {
    let maxReach: number = 0;
    
    for (let i = 0; i <= maxReach && i < nums.length; i++) {
        maxReach = Math.max(maxReach, i + nums[i]);
        if (maxReach >= nums.length - 1) {
            return true;
        }
    }
    
    return false;
}

File: 55. Jump Game/55. Jump Game.rs
impl Solution {
    pub fn can_jump(nums: Vec<i32>) -> bool {
        let mut max_reach = 0;
        
        for (i, &jump) in nums.iter().enumerate() {
            if i > max_reach {
                return false;
            }
            max_reach = max_reach.max(i + jump as usize);
            if max_reach >= nums.len() - 1 {
                return true;
            }
        }
        
        true
    }
}